### PR TITLE
Benchmark cleanup

### PR DESCRIPTION
**Pull request by roji**
This PR cleans up some benchmark stuff.

@sebastienros once this is merged, we'd need the corresponding cleanup to also occur in the database. I hope you can help us with that...

Benchmark name changes:

* InitializationSqliteTests -> Initialization
* FuncletizationSqliteTests -> Funcletization
* NavigationsQuerySqliteTests -> NavigationsQuery (does not appear in dashboard!)
* QueryCompilationSqliteTests -> QueryCompilation (does not appear in dashboard!)
* RawSqlQuerySqliteTests -> RawSqlQuery
* SimpleQuerySqliteTests -> SimpleQuery

For the following, I've configured BDN to always show the times for exactly one operation, even when the benchmark internally executes a loop. That means we'd need to multiply historical values in the database:

* AddDataVariations: multiply by 20000
* ExistingDataVariations: multiply by 20000
* ChildVariations: multiply by 10000
* ParentVariations: multiply by 5000
* InitializationTests.CreateAndDisposeUnusedContext: multiply by 10000
* InitializationTests.InitializeAndQuery_AdventureWorks: multiply by 1000
* InitializationTests.InitializeAndSaveChanges_AdventureWorks: multiply by 100
* FuncletizationTests: multiply by 100
* NavigationsQueryTests: multiply by 10
* QueryCompilation: multiply by 10

### CODE REVIEW

**Comment by smitpatel** [comment]
I don't think OperationsPerInvoke is correct thing for all of them. Bring it to design meeting please.

File: benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs [hunk 1]
```diff
@@ -65,13 +67,13 @@ public virtual void Add()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
             public virtual void AddRange()
             {
                 _context.Customers.AddRange(_customersWithoutPk);
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs:73]: AddRange using OPI

File: benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs [hunk 2]
```diff
@@ -80,7 +82,7 @@ public virtual void Attach()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
             public virtual void AttachRange()
             {
                 _context.Customers.AttachRange(_customersWithPk);
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs:88]: AddRange using OPI

File: benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs [hunk 3]
```diff
@@ -105,13 +107,13 @@ public virtual void Remove()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
             public virtual void RemoveRange()
             {
                 _context.Customers.RemoveRange(_customersWithPk);
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs:113]: RemoveRange using OPI

File: benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs [hunk 4]
```diff
@@ -120,7 +122,7 @@ public virtual void Update()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
             public virtual void UpdateRange()
             {
                 _context.Customers.UpdateRange(_customersWithPk);
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/DbSetOperationTests.cs:128]: UpdateRange using OPI

File: benchmark/EFCore.Benchmarks/ChangeTracker/FixupTests.cs [hunk 5]
```diff
@@ -86,7 +90,7 @@ public virtual void AttachChildren()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = TotalOrders)]
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/FixupTests.cs:93]: ???

File: benchmark/EFCore.Benchmarks/ChangeTracker/FixupTests.cs [hunk 6]
```diff
@@ -120,7 +124,7 @@ public virtual void AttachParents()
                 }
             }
 
-            [Benchmark]
+            [Benchmark(OperationsPerInvoke = Customers)]
```
Comment (smitpatel) [inline benchmark/EFCore.Benchmarks/ChangeTracker/FixupTests.cs:127]: ???

**Comment by roji** [comment]
OK we've discussed this in design and agreed it's a good direction.

@sebastienros, this now depends on you :) When we merge this, all the above changes will need to take place in the database - will you be able to help us with this?

**Comment by smitpatel** [comment]
Ping!

**Comment by sebastienros** [comment]
Not yet ready to do it. But this will be done.

**Comment by roji** [comment]
Closing for age, will re-tackle this again in the future. Tracking issues are #25361 and #19874.

**Pull request closed without merging**